src/smc-webapp/jupyter/actions.coffee

```diff
@@ -1346,14 +1346,22 @@ class exports.JupyterActions extends Actions
                 @_fetching_backend_kernel_info = false
 
     # Do a file action, e.g., 'compress', 'delete', 'rename', 'duplicate', 'move',
-    # 'copy', 'share', 'download', 'open_file', 'close_file'.
+    # 'copy', 'share', 'download', 'open_file', 'close_file', 'reopen_file'
     # Each just shows
     # the corresponding dialog in
     # the file manager, so gives a step to confirm, etc.
     # The path may optionally be *any* file in this project.
     file_action: (action_name, path) =>
         a = @redux.getProjectActions(@store.get('project_id'))
         path ?= @store.get('path')
+        if action_name == 'reopen_file'
+            a.close_file(path)
+            # ensure the side effects from changing registered
+            # editors in project_file.coffee finish happening
+            window.setTimeout =>
+                a.open_file(path: path)
```

Comment:
I think it's interesting that this seems necessary... If this is the real fix then the current code shouldn't work at all. Maybe it has something to do with how `require` is resolved by upgrading webpack.